eclipse-collections/src/main/java/org/eclipse/collections/impl/bimap/mutable/UnmodifiableBiMap.java

```diff
@@ -0,0 +1,1119 @@
+    }
+
+    @Override
+    public void putAll(Map<? extends K, ? extends V> m)
+    {
+        throw new UnsupportedOperationException("Cannot call putAll() on " + this.getClass().getSimpleName());
+    }
+
+    @Override
+    public void clear()
+    {
+        throw new UnsupportedOperationException("Cannot call clear() on " + this.getClass().getSimpleName());
+    }
+
+    @Override
+    public Set<K> keySet()
+    {
+        return Collections.unmodifiableSet(this.delegate.keySet());
+    }
+
+    @Override
+    public Collection<V> values()
+    {
+        return Collections.unmodifiableCollection(this.delegate.values());
+    }
+
+    @Override
+    public Set<Entry<K, V>> entrySet()
+    {
+        return Collections.unmodifiableMap(this.delegate).entrySet();
+    }
+
+    @Override
+    public V forcePut(K key, V value)
+    {
+        throw new UnsupportedOperationException("Cannot call forcePut() on " + this.getClass().getSimpleName());
+    }
+
+    @Override
+    public MutableBiMap<K, V> asSynchronized()
+    {
+        return this.delegate.asSynchronized();
+    }
+
+    @Override
+    public MutableBiMap<K, V> clone()
+    {
+        return this;
+    }
+
+    @Override
+    public MutableBiMap<K, V> asUnmodifiable()
+    {
+        return this;
+    }
+
+    @Override
+    public V get(Object key)
+    {
+        return this.delegate.get(key);
+    }
+
+    @Override
+    public boolean containsKey(Object key)
+    {
+        return this.delegate.containsKey(key);
+    }
+
+    @Override
+    public boolean containsValue(Object value)
+    {
+        return this.delegate.containsValue(value);
+    }
+
+    @Override
+    public void forEachValue(Procedure<? super V> procedure)
+    {
+        this.delegate.forEachValue(procedure);
+    }
+
+    @Override
+    public int size()
+    {
+        return this.delegate.size();
+    }
+
+    @Override
+    public boolean isEmpty()
+    {
+        return this.delegate.isEmpty();
+    }
+
+    @Override
+    public boolean notEmpty()
+    {
+        return this.delegate.notEmpty();
+    }
+
+    @Override
+    public V getFirst()
+    {
+        return this.delegate.getFirst();
+    }
+
+    @Override
+    public V getLast()
+    {
+        return this.delegate.getLast();
+    }
+
+    @Override
+    public boolean contains(Object object)
+    {
+        return this.delegate.contains(object);
+    }
+
+    @Override
+    public boolean containsAllIterable(Iterable<?> source)
+    {
+        return this.delegate.containsAllIterable(source);
+    }
+
+    @Override
+    public boolean containsAll(Collection<?> source)
+    {
+        return this.delegate.containsAll(source);
+    }
+
+    @Override
+    public boolean containsAllArguments(Object... elements)
+    {
+        return this.delegate.containsAllArguments(elements);
+    }
+
+    @Override
+    public MutableBiMap<K, V> tap(Procedure<? super V> procedure)
```

Comment:
tap() is an unusual method when it comes to wrappers, so I think it should be higher up in the file before most other iteration patterns.